schema/snapcraft.json

```diff
@@ -568,7 +568,7 @@
                                 "legacy",
                                 "full"
                             ],
-                            "default": "legacy"
+                            "default": "full"
```

Comment:
this entire entry should be invalid for anything NOT core or core18, changing this is not desired.

Reply:
In master, there's no special handling AFAICT for "full" or "legacy".

I went back and looked at the behavior for 3.8 to compare:

```
if adapter == "full":
   command: <command>
   command-chain:
   - snap/command-chain/snapcraft-runner

elif adapter == "legacy":
   command: snap/command-chain/snapcraft-runner <command>

elif adapter == "none":`
  command: <command>
```

The current behavior I see (with this PR anyhow) is that if the adapter is "none", command-chain usage is disabled.

In master, it always uses command-chain.  Was that not intentional?  And doesn't that effectively mean "full" is synonymous with "legacy"? 

Should:
(1) the old legacy behavior be restored and
(2) and maintained as default?

Reply:
FWIW, testing with _classic_ confinement:

Stable 3.8:

```
if adapter == "full":
   command: <command>
   command-chain:
   - snap/command-chain/snapcraft-runner

elif adapter == "legacy":
   command: snap/command-chain/snapcraft-runner command-<command>.wrapper

elif adapter == "none":
  command: <command>
```

In master, today without this PR:

```
if adapter == "full":
   command: <command>

elif adapter == "legacy":
   command: <command>

elif adapter == "none":
   command: <command>
```

Reply:
Look at the contents of the snapcraft-runner, it should be empty for classic, more so, not something snapcraft should set. 3.8 is probably buggy.

Reply:
Did you see a more extensive post prior to this?  It seems to be gone?

Reply:
Was in my email, but not showing on here...

```
In master, there's no special handling AFAICT for "full" or "legacy".

I went back and looked at the behavior for 3.8 to compare:

if adapter == "full":
   command: <command>
   command-chain:
   - snap/command-chain/snapcraft-runner

elif adapter == "legacy":
   command: snap/command-chain/snapcraft-runner <command>

elif adapter == "none":`
  command: <command>

The current behavior I see (with this PR anyhow) is that if the adapter is "none", command-chain usage is disabled.

In master, it always uses command-chain. Was that not intentional? And doesn't that effectively mean "full" is synonymous with "legacy"?

Should:
(1) the old legacy behavior be restored and
(2) and maintained as default?
```